### PR TITLE
Added basic auth support for Mesos provider.

### PR DESCRIPTION
**Pull request by highfly22**
This patch works with Marathon configured basic authentication.

### CODE REVIEW

**Comment by highfly22** [comment]
I signed it!

**Comment by highfly22** [comment]
I signed it!

File: perfkitbenchmarker/providers/mesos/mesos_docker_instance.py [hunk 1]
```diff
@@ -126,7 +132,7 @@ def _CreateApp(self):
     logging.info("Attempting to create App: %s" % self.name)
     body = self._BuildAppBody()
     headers = {'content-type': 'application/json'}
-    output = requests.post(self.api_url, data=body, headers=headers)
+    output = requests.post(self.api_url, data=body, headers=headers, auth=self.auth)
```
Comment (tedsta) [inline perfkitbenchmarker/providers/mesos/mesos_docker_instance.py]: Linter error here because line is too long

Comment (highfly22) [inline perfkitbenchmarker/providers/mesos/mesos_docker_instance.py]: Fix the lint error by rewrite git log.

**Pull request merged**